src/input/compact/InputSettlerCompact.sol

```diff
@@ -54,10 +54,26 @@ contract InputSettlerCompact is InputSettlerPurchase, IInputSettlerCompact {
 
     constructor(
         address compact
-    ) EIP712("OIFCompact", "1") {
+    ) EIP712(_domainName(), _domainVersion()) {
         COMPACT = TheCompact(compact);
     }
 
+    /**
+     * @notice Returns the domain name of the EIP712 signature.
+     * @return name The domain name.
+     */
+    function _domainName() internal pure override returns (string memory) {
```

Comment:
Very important to make these functions `virtual`, otherwise the issue isn't solved.